frontends/onnx-frontend/third_party/patches/OnnxMlirPooling.patch

```diff
@@ -0,0 +1,136 @@
++
++    auto inputTy = cast<RankedTensorType>(input.getType());
++    auto outTy = cast<RankedTensorType>(cumSumOp.getType());
++    auto inputElemTy = inputTy.getElementType();
++    auto inputRank = inputTy.getRank();
++    auto inputShape = inputTy.getShape();
++
++    Value initVal = createInitialValueForPoolingOp(op, inputElemTy, rewriter);
++    SmallVector<int64_t> stablehloKernelSize(inputRank, 1);
++    DenseIntElementsAttr dimAttr;
++    if (!matchPattern(axis, m_Constant(&dimAttr))) {
++      return op->emitError("Not support dynamic axis yet.");
++    }
++    int64_t dim = dimAttr.getValues<APInt>()[0].getSExtValue();
++    dim = positiveDim(dim, inputRank);
++
++    if (inputTy.isDynamicDim(dim)) {
++      return op->emitError("Not support dynamic dim yet.");
++    }
++
++    stablehloKernelSize[dim] = inputShape[dim];
++    SmallVector<int64_t> stablehloStride(inputRank, 1);
++    SmallVector<int64_t> stablehloDilation(inputRank, 1);
++    SmallVector<int64_t> stablehloPadding(inputRank * 2, 0);
++    stablehloPadding[dim * 2] = inputShape[dim] - 1;
++
++    auto windowDimensions = rewriter.getI64TensorAttr(stablehloKernelSize);
++    auto windowStrides = rewriter.getI64TensorAttr(stablehloStride);
++    DenseIntElementsAttr baseDilations;
++    auto windowDilations = rewriter.getI64TensorAttr(stablehloDilation);
++    DenseIntElementsAttr pad = DenseIntElementsAttr::get(
++        RankedTensorType::get(
++            {static_cast<int64_t>(inputRank), static_cast<int64_t>(2)},
++            rewriter.getI64Type()),
++        stablehloPadding);
++
++    auto reduceWindowSum = rewriter.create<stablehlo::ReduceWindowOp>(
++        op->getLoc(), outTy, input, initVal, windowDimensions, windowStrides,
++        baseDilations, windowDilations, pad);
++
++    Block &sumBlock = reduceWindowSum.getBody().emplaceBlock();
++
++    // Add arguments
++    auto blockArgumentType = RankedTensorType::get({}, inputElemTy);
++    sumBlock.addArgument(blockArgumentType, op->getLoc());
++    sumBlock.addArgument(blockArgumentType, op->getLoc());
++    auto *firstArg = sumBlock.args_begin();
++    auto *secondArg = std::next(firstArg);
++
++    {
++      OpBuilder::InsertionGuard guard(rewriter);
++      rewriter.setInsertionPointToStart(&sumBlock);
++
++      Value sumResult = rewriter.create<stablehlo::AddOp>(
++          op->getLoc(), *firstArg, *secondArg);
++      rewriter.create<stablehlo::ReturnOp>(op->getLoc(), sumResult);
++    }
++
++    rewriter.replaceOp(op, reduceWindowSum.getResults());
++    return success();
++  }
++};
++
+ } // namespace
+ 
+ void populateLoweringONNXPoolingOpToStablehloPattern(
+@@ -223,6 +329,7 @@ void populateLoweringONNXPoolingOpToStablehloPattern(
+       ONNXMaxPoolSingleOutOpAdaptor, ONNXMaxPoolSingleOutOpShapeHelper>>(ctx);
+   patterns.insert<ONNXPoolOpLoweringToStablehlo<ONNXAveragePoolOp,
+       ONNXAveragePoolOpAdaptor, ONNXAveragePoolOpShapeHelper>>(ctx);
++  patterns.insert<ONNXCumSumOpLowering>(ctx);
```

Comment:
need to add a test for op conversion (In onnx-mlir/test/conversion/onnx_to_stablehlo), add them in the patch, and later make a PR to onnx-mlir upstream